### ISSUE TITLE
Unable to install any app on Saleor after recent release of saleor-apps: Issues with /api/register endpoint.

### ISSUE THREAD

**Issue by wackyesolution** [saleor/apps#340]
Since the recent release of saleor-apps, I am unable to install any app on my Saleor backend. I followed the common installation guide, deployed the app, set the env variables, uploaded the app file, and passed the app's manifest to the dashboard. However, every time I pass the manifest of an app and try to install, I receive the following error: "The auth data given during the registration request could not be used to fetch the app ID". Moreover, no .saleor-app-auth.json file is created in the application folder.

Upon investigation, I realized that all the apps use the /api/register route to register a new Saleor backend. However, every app I deployed returns an HTTP error 405 when I try to access this link using a browser (...except saleor-app-checkout that it allow me to access to /api/register/ route )

I also tried to test the same thing on the application deployed by your team at https://saleor-app-cms-git-main-saleorcommerce.vercel.app/api/register, and you will also see that it does not allow access, and as a result, all the new apps are apparently not accessible.

Another thing I noticed is that the checkout application was the only one able to interface with Saleor API, and I did not understand why only the checkout worked. Upon further investigation, I found that going to the /api/register/ route of the checkout app was the only one that responded! That is why it worked with Saleor without any problems.

In summary:

Why are all the new apps not configurable and returning the error I mentioned above?
Why do all the new apps (cms, mails-and-messages, taxes...) have the /api/register route that rejects all connections?
Is it a coincidence that the checkout can be configured correctly, and if you go to the app deployed by the team -> https://saleor-app-checkout-git-main-saleorcommerce.vercel.app/api/register/, you can see that the browser responds while if you go to https://saleor-app-cms-git-main-saleorcommerce.vercel.app/api/register/, you are immediately dropped out?

**Comment by lkostrowski** [comment]
Hey, looks like everything works as expected.

First, `api/register` route returns `405` because its invalid method. Browser call only allow GET endpoint, but you have to call `POST` method - if you try it from postman/insomnia/curl, you will see that endpoint will pass.

Second, this endpoint shouldn't be called by yourself, its called by Saleor during installation. Its a POST request with body containing app tokens (sort of handshake). It doesn't make any sens to call it yourself, its internal installation process between Saleor and App

Every app that Saleor maintains use or will use sdk's `createRegisterHandler` to abstract connection internals. We will check why checkout doesn't have it implemented.

However, this may be an actual issue
```
I receive the following error: "The auth data given during the registration request could not be used to fetch the app ID". Moreover, no .saleor-app-auth.json file is created in the application folder.
```

But we need more info here. 
Please check [APL docs](https://github.com/saleor/saleor-app-sdk/blob/main/docs/apl.md). 
- FileAPL works only locally. 
- Vercel deployments are for internal Saleor usage and they can be protected to be installed only in Saleor cloud

Tip: You can try a free Saleor Cloud account to play with your apps to ensure that problems you have are not related to your self-hosting of Saleor
